lib/hyacinth/adapters/external_identifier_adapter/datacite.rb

```diff
@@ -51,9 +53,10 @@ def exists?(id)
         # @param id [String] the DOI to update
         # @param digital_object [DigitalObject]
         # @param target_url [String] the target URL to be associated with the DOI
-        # @param state [Symbol] doi_state can be set to one of the following: :draft, :findable, :registered
+        # @param publish [Boolean] make the identifier publicly resolvable
         # @return [Boolean] true if update was successful, false otherwise
-        def update_impl(id, digital_object, target_url, state)
+        def update_impl(id, digital_object, target_url, publish = true)
+          state = publish ? :findable : :registered
```

Comment:
As far as DataCite is concerned, it is possible to update a draft DOI to modify the associated metadata and leave it in the draft state. I just want to verify that this use case will never happen in the Hyacinth app. I believe Datacite#update is only called when publishing (resulting state = findable) or unpublishing (resulting state = registered) an item/asset. Is this correct?

Reply:
it is only called when publishing, yes.